Submitted By: Jim Gifford <jim@linuxfromscratch.org>
Date: 2008-07-07
Upstream Status: CLFS Specific
Origin: Jim Gifford
Description: Change /lib/udev to /lib64/udev

diff -Naur udev-124.orig/etc/udev/rules.d/50-udev-default.rules udev-124/etc/udev/rules.d/50-udev-default.rules
--- udev-124.orig/etc/udev/rules.d/50-udev-default.rules	2008-06-12 01:24:30.000000000 -0400
+++ udev-124/etc/udev/rules.d/50-udev-default.rules	2008-07-07 20:29:35.000000000 -0400
@@ -103,5 +103,5 @@
 KERNEL=="iowarrior[0-9]*",	NAME="usb/%k"
 
 # do not delete static device nodes
-ACTION=="remove", NAME=="?*", TEST=="/lib/udev/devices/$name", OPTIONS+="ignore_remove"
+ACTION=="remove", NAME=="?*", TEST=="/lib64/udev/devices/$name", OPTIONS+="ignore_remove"
 
diff -Naur udev-124.orig/extras/ata_id/Makefile udev-124/extras/ata_id/Makefile
--- udev-124.orig/extras/ata_id/Makefile	2008-06-12 01:24:30.000000000 -0400
+++ udev-124/extras/ata_id/Makefile	2008-07-07 20:29:35.000000000 -0400
@@ -16,7 +16,7 @@
 sbindir =	${prefix}/sbin
 usrbindir =	${prefix}/usr/bin
 usrsbindir =	${prefix}/usr/sbin
-libudevdir =	${prefix}/lib/udev
+libudevdir =	${prefix}/lib64/udev
 mandir =	${prefix}/usr/share/man
 configdir =	${etcdir}/udev/
 
diff -Naur udev-124.orig/extras/cdrom_id/Makefile udev-124/extras/cdrom_id/Makefile
--- udev-124.orig/extras/cdrom_id/Makefile	2008-06-12 01:24:30.000000000 -0400
+++ udev-124/extras/cdrom_id/Makefile	2008-07-07 20:29:35.000000000 -0400
@@ -16,7 +16,7 @@
 sbindir =	${prefix}/sbin
 usrbindir =	${prefix}/usr/bin
 usrsbindir =	${prefix}/usr/sbin
-libudevdir =	${prefix}/lib/udev
+libudevdir =	${prefix}/lib64/udev
 mandir =	${prefix}/usr/share/man
 configdir =	${etcdir}/udev
 
diff -Naur udev-124.orig/extras/collect/Makefile udev-124/extras/collect/Makefile
--- udev-124.orig/extras/collect/Makefile	2008-06-12 01:24:30.000000000 -0400
+++ udev-124/extras/collect/Makefile	2008-07-07 20:29:35.000000000 -0400
@@ -11,7 +11,7 @@
 sbindir =	${prefix}/sbin
 usrbindir =	${prefix}/usr/bin
 usrsbindir =	${prefix}/usr/sbin
-libudevdir =	${prefix}/lib/udev
+libudevdir =	${prefix}/lib64/udev
 mandir =	${prefix}/usr/share/man
 configdir =	${etcdir}/udev/
 
diff -Naur udev-124.orig/extras/edd_id/Makefile udev-124/extras/edd_id/Makefile
--- udev-124.orig/extras/edd_id/Makefile	2008-06-12 01:24:30.000000000 -0400
+++ udev-124/extras/edd_id/Makefile	2008-07-07 20:29:35.000000000 -0400
@@ -16,7 +16,7 @@
 sbindir =	${prefix}/sbin
 usrbindir =	${prefix}/usr/bin
 usrsbindir =	${prefix}/usr/sbin
-libudevdir =	${prefix}/lib/udev
+libudevdir =	${prefix}/lib64/udev
 mandir =	${prefix}/usr/share/man
 configdir =	${etcdir}/udev/
 
diff -Naur udev-124.orig/extras/firmware/firmware.sh udev-124/extras/firmware/firmware.sh
--- udev-124.orig/extras/firmware/firmware.sh	2008-06-12 01:24:30.000000000 -0400
+++ udev-124/extras/firmware/firmware.sh	2008-07-07 20:29:35.000000000 -0400
@@ -1,6 +1,6 @@
 #!/bin/sh -e
 
-FIRMWARE_DIRS="/lib/firmware /usr/local/lib/firmware"
+FIRMWARE_DIRS="/lib64/firmware /lib/firmware /usr/local/lib64/firmware /usr/local/lib/firmware"
 
 err() {
 	echo "$@" >&2
diff -Naur udev-124.orig/extras/firmware/Makefile udev-124/extras/firmware/Makefile
--- udev-124.orig/extras/firmware/Makefile	2008-06-12 01:24:30.000000000 -0400
+++ udev-124/extras/firmware/Makefile	2008-07-07 20:29:35.000000000 -0400
@@ -13,7 +13,7 @@
 sbindir =	${prefix}/sbin
 usrbindir =	${prefix}/usr/bin
 usrsbindir =	${prefix}/usr/sbin
-libudevdir =	${prefix}/lib/udev
+libudevdir =	${prefix}/lib64/udev
 mandir =	${prefix}/usr/share/man
 configdir =	${etcdir}/udev/
 
diff -Naur udev-124.orig/extras/floppy/Makefile udev-124/extras/floppy/Makefile
--- udev-124.orig/extras/floppy/Makefile	2008-06-12 01:24:30.000000000 -0400
+++ udev-124/extras/floppy/Makefile	2008-07-07 20:29:35.000000000 -0400
@@ -16,7 +16,7 @@
 sbindir =	${prefix}/sbin
 usrbindir =	${prefix}/usr/bin
 usrsbindir =	${prefix}/usr/sbin
-libudevdir =	${prefix}/lib/udev
+libudevdir =	${prefix}/lib64/udev
 mandir =	${prefix}/usr/share/man
 configdir =	${etcdir}/udev/
 
diff -Naur udev-124.orig/extras/fstab_import/Makefile udev-124/extras/fstab_import/Makefile
--- udev-124.orig/extras/fstab_import/Makefile	2008-06-12 01:24:30.000000000 -0400
+++ udev-124/extras/fstab_import/Makefile	2008-07-07 20:29:35.000000000 -0400
@@ -16,7 +16,7 @@
 sbindir =	${prefix}/sbin
 usrbindir =	${prefix}/usr/bin
 usrsbindir =	${prefix}/usr/sbin
-libudevdir =	${prefix}/lib/udev
+libudevdir =	${prefix}/lib64/udev
 mandir =	${prefix}/usr/share/man
 configdir =	${etcdir}/udev/
 
diff -Naur udev-124.orig/extras/path_id/Makefile udev-124/extras/path_id/Makefile
--- udev-124.orig/extras/path_id/Makefile	2008-06-12 01:24:30.000000000 -0400
+++ udev-124/extras/path_id/Makefile	2008-07-07 20:29:35.000000000 -0400
@@ -13,7 +13,7 @@
 sbindir =	${prefix}/sbin
 usrbindir =	${prefix}/usr/bin
 usrsbindir =	${prefix}/usr/sbin
-libudevdir =	${prefix}/lib/udev
+libudevdir =	${prefix}/lib64/udev
 mandir =	${prefix}/usr/share/man
 configdir =	${etcdir}/udev/
 
diff -Naur udev-124.orig/extras/rule_generator/Makefile udev-124/extras/rule_generator/Makefile
--- udev-124.orig/extras/rule_generator/Makefile	2008-06-12 01:24:30.000000000 -0400
+++ udev-124/extras/rule_generator/Makefile	2008-07-07 20:29:35.000000000 -0400
@@ -13,7 +13,7 @@
 sbindir =	${prefix}/sbin
 usrbindir =	${prefix}/usr/bin
 usrsbindir =	${prefix}/usr/sbin
-libudevdir =	${prefix}/lib/udev
+libudevdir =	${prefix}/lib64/udev
 mandir =	${prefix}/usr/share/man
 configdir =	${etcdir}/udev
 
diff -Naur udev-124.orig/extras/rule_generator/write_cd_rules udev-124/extras/rule_generator/write_cd_rules
--- udev-124.orig/extras/rule_generator/write_cd_rules	2008-06-12 01:24:30.000000000 -0400
+++ udev-124/extras/rule_generator/write_cd_rules	2008-07-07 20:29:35.000000000 -0400
@@ -13,7 +13,7 @@
 
 RULES_FILE="/etc/udev/rules.d/70-persistent-cd.rules"
 
-. /lib/udev/rule_generator.functions
+. /lib64/udev/rule_generator.functions
 
 find_next_available() {
 	raw_find_next_available "$(find_all_rules 'SYMLINK+=' "$1")"
diff -Naur udev-124.orig/extras/rule_generator/write_net_rules udev-124/extras/rule_generator/write_net_rules
--- udev-124.orig/extras/rule_generator/write_net_rules	2008-06-12 01:24:30.000000000 -0400
+++ udev-124/extras/rule_generator/write_net_rules	2008-07-07 20:29:35.000000000 -0400
@@ -24,7 +24,7 @@
 
 RULES_FILE='/etc/udev/rules.d/70-persistent-net.rules'
 
-. /lib/udev/rule_generator.functions
+. /lib64/udev/rule_generator.functions
 
 interface_name_taken() {
 	local value="$(find_all_rules 'NAME=' $INTERFACE)"
diff -Naur udev-124.orig/extras/scsi_id/Makefile udev-124/extras/scsi_id/Makefile
--- udev-124.orig/extras/scsi_id/Makefile	2008-06-12 01:24:30.000000000 -0400
+++ udev-124/extras/scsi_id/Makefile	2008-07-07 20:29:35.000000000 -0400
@@ -19,7 +19,7 @@
 sbindir =	${prefix}/sbin
 usrbindir =	${prefix}/usr/bin
 usrsbindir =	${prefix}/usr/sbin
-libudevdir =	${prefix}/lib/udev
+libudevdir =	${prefix}/lib64/udev
 mandir =	${prefix}/usr/share/man
 configdir =	${etcdir}/udev/
 
diff -Naur udev-124.orig/extras/usb_id/Makefile udev-124/extras/usb_id/Makefile
--- udev-124.orig/extras/usb_id/Makefile	2008-06-12 01:24:30.000000000 -0400
+++ udev-124/extras/usb_id/Makefile	2008-07-07 20:29:35.000000000 -0400
@@ -16,7 +16,7 @@
 sbindir =	${prefix}/sbin
 usrbindir =	${prefix}/usr/bin
 usrsbindir =	${prefix}/usr/sbin
-libudevdir =	${prefix}/lib/udev
+libudevdir =	${prefix}/lib64/udev
 mandir =	${prefix}/usr/share/man
 configdir =	${etcdir}/udev/
 
diff -Naur udev-124.orig/extras/volume_id/lib/Makefile udev-124/extras/volume_id/lib/Makefile
--- udev-124.orig/extras/volume_id/lib/Makefile	2008-06-12 01:24:30.000000000 -0400
+++ udev-124/extras/volume_id/lib/Makefile	2008-07-07 20:31:21.000000000 -0400
@@ -5,8 +5,8 @@
 # Released under the GNU General Public License, version 2.
 #
 includedir = 	${prefix}/usr/include
-libdir =	${prefix}/lib
-usrlibdir =	${prefix}/usr/lib
+libdir =	${prefix}/lib64
+usrlibdir =	${prefix}/usr/lib64
 
 INSTALL = install -c
 INSTALL_DATA  = ${INSTALL} -m 644
diff -Naur udev-124.orig/extras/volume_id/Makefile udev-124/extras/volume_id/Makefile
--- udev-124.orig/extras/volume_id/Makefile	2008-06-12 01:24:30.000000000 -0400
+++ udev-124/extras/volume_id/Makefile	2008-07-07 20:29:35.000000000 -0400
@@ -14,7 +14,7 @@
 sbindir =	${prefix}/sbin
 usrbindir =	${prefix}/usr/bin
 usrsbindir =	${prefix}/usr/sbin
-libudevdir =	${prefix}/lib/udev
+libudevdir =	${prefix}/lib64/udev
 mandir =	${prefix}/usr/share/man
 configdir =	${etcdir}/udev/
 
diff -Naur udev-124.orig/Makefile udev-124/Makefile
--- udev-124.orig/Makefile	2008-06-12 01:24:30.000000000 -0400
+++ udev-124/Makefile	2008-07-07 20:29:35.000000000 -0400
@@ -88,7 +88,7 @@
 etcdir =	${prefix}/etc
 sbindir =	${prefix}/sbin
 usrbindir =	${prefix}/usr/bin
-libudevdir =	${prefix}/lib/udev
+libudevdir =	${prefix}/lib64/udev
 mandir =	${prefix}/usr/share/man
 configdir =	${etcdir}/udev
 udevdir =	/dev
diff -Naur udev-124.orig/udev_rules.c udev-124/udev_rules.c
--- udev-124.orig/udev_rules.c	2008-06-12 01:24:30.000000000 -0400
+++ udev-124/udev_rules.c	2008-07-07 20:29:35.000000000 -0400
@@ -172,9 +172,9 @@
 		}
 	}
 
-	/* allow programs in /lib/udev called without the path */
+	/* allow programs in /lib64/udev called without the path */
 	if (strchr(argv[0], '/') == NULL) {
-		strlcpy(program, "/lib/udev/", sizeof(program));
+		strlcpy(program, "/lib64/udev/", sizeof(program));
 		strlcat(program, argv[0], sizeof(program));
 		argv[0] = program;
 	}
diff -Naur udev-124.orig/udev_rules_parse.c udev-124/udev_rules_parse.c
--- udev-124.orig/udev_rules_parse.c	2008-06-12 01:24:30.000000000 -0400
+++ udev-124/udev_rules_parse.c	2008-07-07 20:29:35.000000000 -0400
@@ -447,9 +447,9 @@
 				if (pos)
 					pos[0] = '\0';
 
-				/* allow programs in /lib/udev called without the path */
+				/* allow programs in /lib64/udev called without the path */
 				if (strchr(file, '/') == NULL) {
-					strlcpy(file, "/lib/udev/", sizeof(file));
+					strlcpy(file, "/lib64/udev/", sizeof(file));
 					strlcat(file, value, sizeof(file));
 					pos = strchr(file, ' ');
 					if (pos)
